### PR TITLE
Update maintainers and bump pytz

### PR DESCRIPTION
**Pull request by BenWilson2**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by BenWilson2** [comment]
@conda-forge-admin, please rerender

**Comment by harupy** [comment]
@xhochy Could you also take a look? Thanks in advance!

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1 +1 @@
-* @BenWilson2 @WeichenXu123 @aarondav @ahirreddy @andrewmchen @aveshcsingh @dbczumar @harupy @janjagusch @jaroslawk @mateiz @mparkhe @pogil @smurching @sueann @tomasatdatabricks @xhochy @zangr
\ No newline at end of file
+* @BenWilson2 @WeichenXu123 @dbczumar @harupy
```
Comment (harupy) [inline .github/CODEOWNERS:1]: We can't remove the conda-forge members.

**Comment by BenWilson2** [comment]
@xhochy are the changes in this PR valid for us to remove the large list of maintainers?

**Comment by xhochy** [comment]
Only do these changes in meta.yaml, all other files are auto-generated. Also, make sure to only remove people that were added here because they were added as they were working on it on behalf of Databricks.

**Pull request closed without merging**